databases/functions/users/get-users-karma-score.js

```diff
@@ -0,0 +1,19 @@
+/**
+ *
+ * @param {import('sequelize').Model} userModel
+ * @param {String} userId
+ */
+const getUsersKarmaScore = async (userModel, userIds = []) => {
+  if (userIds.length === 0) return [];
+  const users = await userModel.findAll({
+    where: {
+      user_id: userIds
+    },
+    attributes: ['user_id', 'karma_score'],
+    raw: true
+  });
+
+  return users;
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `getUsersKarmaScore` is well implemented. It checks if the `userIds` array is empty and returns an empty array if true, which is a good practice to avoid unnecessary database queries. However, it's important to note that this function will return an empty array if no users are found with the provided IDs. If this behavior is not intended, you might want to add error handling for such cases.

```diff
  const users = await userModel.findAll({
    where: {
      user_id: userIds
    },
    attributes: ['user_id', 'karma_score'],
    raw: true
  });

+ if (users.length === 0) {
+   throw new Error('No users found with the provided IDs');
+ }

  return users;
```
This way, the caller of the function can handle this error appropriately, depending on the context in which this function is used.


<!-- This is an auto-generated comment by OSS CodeRabbit -->